### PR TITLE
SNOW-1989063: Fix reading tokens in Windows Credential Manager implementation

### PR DESCRIPTION
**Pull request by sfc-gh-ext-simba-lf**
### Description
Fixes reading too many chars from Windows native secret store tokens

### Checklist
- [ ] Code compiles correctly
- [ ] Code is formatted according to [Coding Conventions](../blob/master/CodingConventions.md)
- [ ] Created tests which fail without the change (if possible)
- [ ] All tests passing (`dotnet test`)
- [ ] Extended the README / documentation, if necessary
- [ ] Provide JIRA issue id (if possible) or GitHub issue id in PR name

### CODE REVIEW

File: Snowflake.Data/Core/CredentialManager/Infrastructure/SFCredentialManagerWindowsNativeImpl.cs [hunk 1]
```diff
@@ -46,6 +46,10 @@ public string GetCredentials(string key)
             using (var critCred = new CriticalCredentialHandle(nCredPtr))
             {
                 var cred = critCred.GetCredential();
+                if (cred.CredentialBlob.Length > cred.CredentialBlobSize / 2)
```
Comment (sfc-gh-knozderko) [inline Snowflake.Data/Core/CredentialManager/Infrastructure/SFCredentialManagerWindowsNativeImpl.cs:49]: could you test if there would not be any null pointer exception from `cred.CredentialBlob.Length` in case if the entry was not found (maybe cred.CredentialBlob could be null)?

Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data/Core/CredentialManager/Infrastructure/SFCredentialManagerWindowsNativeImpl.cs:49]: [If the entry is not found](https://github.com/snowflakedb/snowflake-connector-net/pull/1116/files#diff-f72b20c0757b2d9508bd759aec27ac4b1bab3205a38a2d62714c25d4e7d4135aR40) then it would not reach this part of the code

I added another test that calls GetCredentials() for a key that doesn't exist to check that it returns an empty string

Comment (sfc-gh-knozderko) [inline Snowflake.Data/Core/CredentialManager/Infrastructure/SFCredentialManagerWindowsNativeImpl.cs:49]: maybe here you could also cast `cred.CredentialBlobSize / 2` to int

**Review by sfc-gh-knozderko — APPROVED** [review]

**Review by sfc-gh-dstempniak — APPROVED** [review]

**Review by sfc-gh-pbulawa — APPROVED** [review]

File: Snowflake.Data/Core/CredentialManager/Infrastructure/SFCredentialManagerWindowsNativeImpl.cs (hunk 1 above)
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data/Core/CredentialManager/Infrastructure/SFCredentialManagerWindowsNativeImpl.cs]: Added cast
